### PR TITLE
Add view with negative dim

### PR DESCRIPTION
**Pull request by eellison**
Stack from [ghstack](https://github.com/ezyang/ghstack):
* #64323 Add embedding shape analysis
* #64121 Max Pool with indices
* #63530 Add Maxpool to shape analysis / Opinfo
* #64300 Add symbolic shape comparison optimization
* #64299 Refactor to use shape arguments
* **#63516 Add view with negative dim**
* #63615 Generalize expand logic
* #63407 Add permute, arange
* #63365 Add support for slice, selec twith int, index_select
* #63099 Add squeeze, unsqueeze, transpose shape functins
* #63050 Add batch of unary functions

how to review: pretty much just check that the inputs generated are a good representation of the op semantics, that should be sufficient for correctness, and then you can also double check the op size semantics by going to https://codebrowser.bddppq.com/pytorch/pytorch/ typing in native::{op_name} and looking at the op implementation as a bonus if you want

Differential Revision: [D30738143](https://our.internmc.facebook.com/intern/diff/D30738143)

### CODE REVIEW

File: test/jit/test_symbolic_shape_analysis.py [hunk 1]
```diff
@@ -91,6 +92,7 @@ def foo(y):
         torch._C._jit_pass_propagate_shapes_on_graph(foo.graph)
         FileCheck().check("Tensor = aten::view").run(foo.graph)
 
+    @unittest.skip("Temp")
```
Comment (eellison) [inline test/jit/test_symbolic_shape_analysis.py:95]: These get removed in a subsequent PR in this stack

**Review by priyaramani — APPROVED** [review]

**Comment by eellison** [comment]
@eellison has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D30738143).

File: test/test_jit_fuser_te.py [hunk 2]
```diff
@@ -2055,6 +2055,8 @@ def bn_neither(i, x):
     # Reference: https://github.com/pytorch/pytorch/pull/59442/checks?check_run_id=2746156896
     't',
     'conj'
+    'view',
```
Comment (eellison) [inline test/test_jit_fuser_te.py:2058]: cc @Chillee who added this test, breaks with negative dim

**Comment by eellison** [comment]
@eellison has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D30738143).

**Comment by eellison** [comment]
@eellison has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D30738143).

**Pull request closed without merging**